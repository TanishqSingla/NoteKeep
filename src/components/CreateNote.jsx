import React, { useState } from "react";

const CreateNote = () => {
  const [note, setNote] = useState({
    title: "",
    content: "",
  });

  const handleChange = (event) => {
    const { name, value } = event.target;
    setNote((prevNote) => {
      return {
        ...prevNote,
        [name]: value,
      };
    });
  };

  return (
    <div>
      <form>
        <input
          name="title"
          type="text"
          onChange={handleChange}
          value={note.title}
          placeholder="Title"
        />
        <textarea
          name="content"
          value={note.content}
          placeholder="Take a note..."
        />
        <button>+</button>
      </form>
    </div>
  );
};

export default CreateNote;
