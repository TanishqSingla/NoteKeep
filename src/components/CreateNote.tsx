import React, { useState } from "react";
import { Notes } from "./App";
import AddIcon from '@material-ui/icons/Add'
interface createNoteProps {
  onAdd: (note: Notes) => void;
}

type onChangeHandler = React.ChangeEvent<
  HTMLInputElement | HTMLTextAreaElement
>;

const CreateNote: React.FC<createNoteProps> = (props) => {
  const [note, setNote] = useState({
    title: "",
    content: "",
  });

  const handleChange = (event: onChangeHandler) => {
    const { name, value } = event.target;
    setNote((prevNote) => {
      return {
        ...prevNote,
        [name]: value,
      };
    });
  };

  const submitNote = (event: React.MouseEvent<HTMLButtonElement>) => {
    props.onAdd(note);
    setNote({
      title: "",
      content: "",
    });
    event.preventDefault();
  };

  return (
    <div>
      <form>
        <input
          name="title"
          type="text"
          onChange={handleChange}
          value={note.title}
          placeholder="Title"
        />
        <textarea
          name="content"
          value={note.content}
          placeholder="Take a note..."
          onChange={handleChange}
        />
        <button onClick={submitNote}><AddIcon /></button>
      </form>
    </div>
  );
};

export default CreateNote;
