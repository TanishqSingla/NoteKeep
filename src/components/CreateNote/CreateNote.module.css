.CreateForm {
  position: relative;
  margin: 30px auto 20px auto;
  padding: 15px;
  width: 480px;
  border-radius: 7px;
  box-shadow: 0 1px 5px #8a8989;
}

.TitleInput{
  width: 100%;
  outline: none;
  border: none;
  font-size: 1.2em;
  resize: none;
  padding: 4px;
}

.AddNoteButton {
  position: absolute;
  display: grid;
  place-items: center;
  height: 36px;
  width: 36px;
  border: none;
  border-radius: 50%;
  color: white;
  bottom: -18px;
  right: 18px;
  cursor: pointer;
  outline: none;
  font-size: 1.2rem;
  background: #222;
}