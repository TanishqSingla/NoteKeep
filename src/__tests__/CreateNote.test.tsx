import { expect, it } from 'vitest'
import renderer from 'react-test-renderer'
import CreateNote from '../components/CreateNote/CreateNote'

it('', () => {
  const component = renderer.create(<CreateNote onAdd={() => console.log('hello')}/>)

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
})