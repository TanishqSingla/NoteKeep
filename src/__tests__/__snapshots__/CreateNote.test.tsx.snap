// Vitest Snapshot v1

exports[` 1`] = `
<div>
  <form>
    <input
      name="title"
      onChange={[Function]}
      placeholder="Title"
      type="text"
      value=""
    />
    <textarea
      name="content"
      onChange={[Function]}
      placeholder="Take a note..."
      value=""
    />
    <button
      onClick={[Function]}
      style={
        {
          "display": "grid",
          "placeItems": "center",
        }
      }
    >
      <style
        dangerouslySetInnerHTML={
          {
            "__html": ".css-i4bv87-MuiSvgIcon-root{-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;width:1em;height:1em;display:inline-block;fill:currentColor;-webkit-flex-shrink:0;-ms-flex-negative:0;flex-shrink:0;-webkit-transition:fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;transition:fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;font-size:1.5rem;}",
          }
        }
        data-emotion="css i4bv87-MuiSvgIcon-root"
      />
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="AddIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
        />
      </svg>
    </button>
  </form>
</div>
`;
